prow/config.gen.yaml

```diff
@@ -447,6 +447,32 @@ presubmits:
         - name: GOCACHE
           value: /tmp/cache
 
+  - name: istio_2.0-gencheck
+    trigger: (?m)^/test( | .* )gencheck,?($|\s.*)
+    decorate: true
+    always_run: true
+    path_alias: istio.io/istio
+    skip_report: false
+    branches:
+      - maistra-2.0
+    rerun_command: /test gencheck
+    spec:
+      containers:
+      - image: "quay.io/maistra-dev/maistra-builder:2.0"
+        imagePullPolicy: Always
+        command:
+        - make
+        - gen-check
```

Comment:
This is great. I actually recently added targets to run the code-gen on our additional `ServiceMesh` bits. I wanted to keep the changes to the upstream `Makefile` minimal though, so it's not run as part of the normal `make gen` target. Can we maybe have this bit read:

```yaml
- make
- maistra-gen-k8s-client
- gen-check
```

That should catch our stuff as well.

Reply:
Will this target just generate files or it's a check that will fail if the PR author didn't generate them? Note that upstream has `gen` and `gen-check`. The latter is meant to fail and to be used in prow.

Reply:
This target just generates the files, but the `gen-check` target works by first running `gen` then running a script that checks that the git working directory is clean. So this should be enough.

Reply:
gotcha, ok, will add that